### PR TITLE
Working on #811: fixed (moved logic inside broadcast receiver)

### PR DESCRIPTION
**Pull request by ognjenns**

### CODE REVIEW

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java [hunk 1]
```diff
@@ -504,8 +505,6 @@ private boolean hasWiredHeadset()
     */
    private void updateAudioDeviceState(boolean hasWiredHeadset)
    {
-      // Update the list of available audio devices.
-      audioDevices.clear();
```
Comment (atsakiridis) [inline restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java]: This seems dangerous; if we keep plugging and unplugging the headset we'll end up with a lot of audioDevices, with unknown results/behavior

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java [hunk 2]
```diff
@@ -403,6 +373,38 @@ private void registerForWiredHeadsetIntentBroadcast()
          @Override
          public void onReceive(Context context, Intent intent)
          {
+
+            // Create and initialize the proximity sensor.
```
Comment (atsakiridis) [inline restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java]: This seems weird, how can we depend initializing stuff like setting up audio focus, etc inside the Receiver that is supposed to fire only when the a wired headset it plugged in or plugged out? Somehow this fires even when there is no headset involved, but I don't think we can trust it; seems easy to break. Isn't there a cleaner way to go?

Also, it's bad practice to create the proximity sensor each time the user plugs in and out the headset. It's lifetime should be the same as the call lifetime.

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java [hunk 3]
```diff
@@ -403,6 +373,38 @@ private void registerForWiredHeadsetIntentBroadcast()
...
+            // Tablet devices (e.g. Nexus 7) does not support proximity sensors.
+            // Note that, the sensor will not be active until start() has been called.
```
Comment (atsakiridis) [inline restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java]: General issue: When testing these changes I'm getting a crash when I start call, plug in and then plug out the wired headset

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java (hunk 2 above)
Comment (ognjenns) [reply restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java]: Ok, lets talk on daily about it.

**Pull request merged**